src/bluez/adapter/mod.rs

```diff
@@ -521,7 +521,7 @@ impl Adapter {
         self.states.contains(&AdapterState::Up)
     }
 
-    pub fn connect(&self) -> Result<ConnectedAdapter> {
-        ConnectedAdapter::new(self)
+    pub fn connect(self) -> Result<ConnectedAdapter> {
+        ConnectedAdapter::new(&self)
     }
```

Comment:
If you rebase this against the latest dev branch the connect method shouldn't be necessary anymore.